quinn-proto/src/shared.rs

```diff
@@ -44,7 +50,7 @@ pub struct TransportConfig {
     pub(crate) minimum_window: u64,
     pub(crate) loss_reduction_factor: u16,
     pub(crate) persistent_congestion_threshold: u32,
-    pub(crate) keep_alive_interval: u32,
+    pub(crate) keep_alive_interval: Option<Duration>,
```

Comment:
I like this, although of course it does make `no_std` harder.

Reply:
Considering the size and number of barriers in the way of `no_std` I'm inclined to not let it get in the way of correctness/ergonomic wins. We can `cfg` in an alternative representation and add a lowest-common-denominator setter (say, `fn keep_alive_interval_ms(&mut self, value: u32)`) if it ends up being needed.

Reply:
Agreed.